### DISCUSSION TITLE
fetch progress information is available but dropped

### DISCUSSION THREAD

**Discussion by jorhett** [gitpython-developers/GitPython#1466] [General]
Modern versions of git will return progress information. In debugging what goes on here, I found that all of the information was being dropped by _parse_line. For others who run into this, I was able to return branch update information to the user by adding the following:

```python
class SeeFetchProgress(git.remote.RemoteProgress):
    # the implementation of RemoteProgress currently doesn't match fetch lines, so catch them when dropped.
    def line_dropped(self, line: str):
        if re.match('remote: ', line):
            return
        print(line)
```

**Comment by Byron** [comment]
Indeed, the mechanism is meant to be used to obtain additional information, with the default implementation being somewhat limited.

This discussion could be used to figure out if the default implementation should be improved, if there should be better docs, or anything else related to improving the situation.

**Reply by jorhett** [reply]
As shown by my change, I believe that the lines other than `remote:` can be useful and necessary... in fact, we wouldn't be able to use this module without the ability to display those lines as they are necessary and critical output to understand the actions taken (or not)

So yes, I firmly believe that this would be much more useful if one could retrieve these lines, hopefully from the API as a String and not by overloading a function.

**Reply by Byron** [reply]
The current architecture should support intercepting all information and transforming it into anything as desired. PRs are definitely welcome.
